### DISCUSSION TITLE
Bad Credentials when creating a check run

### DISCUSSION THREAD

**Discussion by AJNielsen** [community/community#24403] [Apps, API and Webhooks]
<p>I’ve been trying to test out the new GitHub check runs API and ran into a wall. I am able to create a JWT token successfully and use it on the Authenticated App API (<a href="https://developer.github.com/v3/apps/#get-the-authenticated-github-app" rel="nofollow noopener">https://developer.github.com/v3/apps/#get-the-authenticated-github-app</a>) to get a successful response. When I try to call the create a check run API (<a href="https://developer.github.com/v3/checks/runs/#create-a-check-run" rel="nofollow noopener">https://developer.github.com/v3/checks/runs/#create-a-check-run</a>), I am getting a 401 response with the message “Bad Credentials”. I am changing the accept header to the value listed on the create a check run page (have also tried with the accept header from authenticated app api). Looking for some help to figure out where I am going wrong. Should the JWT be modified to access check run endpoints or should the same JWT work?</p>
<p>Followed these directions on authenticating with apps:<br>
<a href="https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/" rel="nofollow noopener">https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/</a></p>
<p>Below is the Raw Request and Response from one of my attempts. As far as I can tell this should work according to the documentation. I’ve also made sure my app has the checks:write permission and is subscribed to the check_suite and check_run webhooks. I am getting data on my server from the check_suite webhook and know that is working.</p>
<p><strong>RAW Request (<em>Omitted some private data</em>)</strong></p>
<p>POST <a href="https://api.github.com/repos/AJNielsen/DarkRelativity/check-runs" rel="nofollow noopener">https://api.github.com/repos/AJNielsen/DarkRelativity/check-runs</a> HTTP/1.1<br>
Authorization: Bearer OMITTED<br>
Content-Type: application/json<br>
Content-Length: 355<br>
Host: <a href="http://api.github.com" rel="nofollow noopener">api.github.com</a><br>
User-Agent: mdtapp001<br>
Accept: application/vnd.github.antiope-preview+json</p>
<p>{“name”:“MyFirstCheck”,“head_sha”:“0054662f1565831624186f590d4dbad87e07bcd6”,“status”:“queued”,“started_at”:“2018-11-01T16:30:00Z”,“output”:{“title”:“MyFirstCheckTitle”,“summary”:“Checking your codes”,“text”:“This is text data.”,“images”:[{“alt”:“A link”,“image_url”:“OMITTED”,“caption”:“A captioned link”}]}}</p>
<p><strong>Raw Response</strong></p>
<p>HTTP/1.1 401 Unauthorized<br>
Date: Fri, 02 Nov 2018 16:16:03 GMT<br>
Content-Type: application/json; charset=utf-8<br>
Content-Length: 83<br>
Server: <a href="http://GitHub.com" rel="nofollow noopener">GitHub.com</a><br>
Status: 401 Unauthorized<br>
X-GitHub-Media-Type: github.antiope-preview; format=json<br>
X-RateLimit-Limit: 60<br>
X-RateLimit-Remaining: 57<br>
X-RateLimit-Reset: 1541178552<br>
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type<br>
Access-Control-Allow-Origin: *<br>
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload<br>
X-Frame-Options: deny<br>
X-Content-Type-Options: nosniff<br>
X-XSS-Protection: 1; mode=block<br>
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin<br>
Content-Security-Policy: default-src ‘none’<br>
X-GitHub-Request-Id: E682:4A9A:5C35:732B:5BDC7838</p>
<p>{“message”:“Bad credentials”,“documentation_url”:“<a href="https://developer.github.com/v3" rel="nofollow noopener">https://developer.github.com/v3</a>”}</p>

**Comment by mathstuf** [comment, marked as answer]
<p>You need to create an <a>installation token</a> using the JWT. That is the token you use to do other API calls. Note that it has an expiration and needs refreshed occasionally.</p>

**Comment by AJNielsen** [comment]
<p>Thanks! Sorry about wasting your time. I realize that I should have made that connection, but I kept reading the API as authenticating as a GitHub App and just got that stuck in my head that it should work. </p>
<p>I went back to read the check run API and it doesn’t mention using installation tokens for auth. It makes a lot of sense that it should be using an installation token especially since I went through installing the app.</p>
<p>Check Run API: <a href="https://developer.github.com/v3/checks/runs/" rel="nofollow noopener">https://developer.github.com/v3/checks/runs/</a></p>

**Comment by peter-kinzelman** [comment]
<p>Meaning after create a installation token via <a href="https://api.github.com/app/installations/%7Binstallation_id%7D/access_tokens" rel="nofollow noopener">https://api.github.com/app/installations/{installation_id}/access_tokens</a> using your previously generated JWT… you’re going to pass that generated token to the /check-runs create endpoint? as Bearer token? </p>
